src/performance-tracking/index.ts

```diff
@@ -0,0 +1,144 @@
+import analytics from '@segment/analytics-react-native';
+// @ts-ignore
+import { SENTRY_ENVIRONMENT } from 'react-native-dotenv';
+import PerformanceMetric from './types/PerformanceMetric';
+import { PerformanceMetricData } from './types/PerformanceMetricData';
+import PerformanceTag from './types/PerformanceTag';
+
+const shouldLogToConsole = __DEV__ || SENTRY_ENVIRONMENT === 'LocalRelease';
+const logTag = '[PERFORMANCE]: ';
+
+function logDurationIfAppropriate(
+  metric: PerformanceMetric,
+  durationInMs: number,
+  ...additionalArgs: any[]
+) {
+  if (shouldLogToConsole) {
+    global.console.log(
+      logTag,
+      `${metric}, duration: ${durationInMs.toFixed(2)}ms`,
+      ...additionalArgs
+    );
+  }
+}
+
+const currentlyTrackedMetrics = new Map<
+  PerformanceMetric,
+  PerformanceMetricData
+>();
+
+interface AdditionalParams extends Record<string, any> {
+  tag?: PerformanceTag;
+}
+
+/**
+ * Function that allows directly commiting performance events.
+ * Useful when we already have duration of something and just want to log it.
+ *
+ * @param metric What you're measuring
+ * @param durationInMs How long did it take
+ * @param additionalParams Any additional context you want to add to your log
+ */
```

Comment:
not a fan of those comments if this is not established as a pattern

Reply:
We have it in many places but I remember seeing them only in JS files. I have mixed feelings about having them in TS as they might get out of sync with typing.

Reply:
Could you expand why is this bad? My reasoning behind using this JSDocs is that IDEs and editors use them when you hover over the text or just trigger parameter suggestions. Since this API is going to be consumed by others I thought it might be useful to add some sort of docs.

How would you want it to be?

Reply:
@tchayen I intentionally didn't add type like `@param {boolean} blah blah blah` It won't get out of sync it will add comment to what certain parameter is

Reply:
Yea, definitely having description is useful. And having short explanation of params is also very helpful.

Regarding to what I'd suggest instead – I have no idea. I feel like mixing JSDoc and TS is not right, but also don't know how to replace that.

Reply:
I think JSDoc in TS is okay. We do this in the design-system, [see here](https://github.com/rainbow-me/rainbow/blob/develop/src/design-system/color/AccentColorContext.tsx#L15). You can also document props by adding `/** */`, like:

```tsx
type ExampleProps = {
  children: ReactNode;
  /** this is some docs on `color` */
  color: string;
}

/**
 * @description This is a description for the Example component.
 */
 function Example(props: ExampleProps) {
```